widlparser/productions.py

```diff
@@ -105,32 +105,37 @@ def _consume_semicolon(self, tokens: Tokenizer, consume_tail: bool = True) -> No
 
 class ComplexProduction(Production):
 	"""
-		Base class for more complex productions;
-		things that can have names, parents, etc.
-		Also, all Constructs.
+	Base class for more complex productions.
+
+	Things that can have names, parents, etc.,
+	including all Constructs.
 	"""
 
-	_parent: Optional[ChildProduction]
+	_parent: Optional[ComplexProduction]
 
 	def __init__(self, tokens: Tokenizer, parent: Optional[ComplexProduction]) -> None:
 		Production.__init__(self, tokens)
 		self._parent = parent
 
 	@property
-	def parent(self) -> ChildProduction:
+	def parent(self) -> ComplexProduction:
 		if self._parent is not None:
 			return self._parent
 		raise KeyError()
 
+	@parent.setter
+	def parent(self, parent: ComplexProduction) -> None:
+		self._parent = parent
+
 	@property
-	def hasParent(self) -> bool:
+	def has_parent(self) -> bool:
 		return self._parent is not None
```

Comment:
```suggestion
		return (self._parent is not None)
```

Reply:
Happy to do the fix, but your style is really to put parens around *all* comparisons? I thought it was just `if` (and presumably `while`) heads.

Reply:
In general, yeah, I'm moving to wrapping every binary operation in ()'s so that later when I add extra clauses the diffs are smaller and less change for getting it wrong.